core/security_group.go

```diff
@@ -0,0 +1,172 @@
+	Name             string   `json:"name,omitempty"`
+	AllowAllInbound  bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound bool     `json:"allow_all_outbound,omitempty"`
+	Associations     []string `json:"associations,omitempty"`
+}
+
+func (sg *SecurityGroup) Ref() SecurityGroupRef {
+	return SecurityGroupRef{ID: sg.ID}
+}
+
+// allows a reference to a security group
+type SecurityGroupRef struct {
+	ID string `json:"id,omitempty"`
+}
+
+func (sg SecurityGroupRef) queryValues() *url.Values {
+	v := &url.Values{}
+	v.Set("security_group[id]", sg.ID)
+
+	return v
+}
+
+type SecurityGroupCreateArguments struct {
+	Name             string    `json:"name,omitempty"`
+	Associations     *[]string `json:"associations,omitempty"`
+	AllowAllInbound  *bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound *bool     `json:"allow_all_outbound,omitempty"`
+}
+
+type SecurityGroupUpdateArguments struct {
+	Name             string    `json:"name,omitempty"`
+	Associations     *[]string `json:"associations,omitempty"`
+	AllowAllInbound  *bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound *bool     `json:"allow_all_outbound,omitempty"`
+}
+
+type SecurityGroupCreateRequest struct {
+	Organization OrganizationRef               `json:"organization"`
+	Properties   *SecurityGroupCreateArguments `json:"properties,omitempty"`
+}
+
+type SecurityGroupUpdateRequest struct {
+	SecurityGroup SecurityGroupRef              `json:"security_group"`
+	Properties    *SecurityGroupUpdateArguments `json:"properties,omitempty"`
+}
+
+type SecurityGroupsResponseBody struct {
+	Pagination     *katapult.Pagination `json:"pagination,omitempty"`
+	SecurityGroup  *SecurityGroup       `json:"security_group,omitempty"`
+	SecurityGroups []*SecurityGroup     `json:"security_groups,omitempty"`
+}
+
+type SecurityGroupsClient struct {
+	client   RequestMaker
+	basePath *url.URL
+}
+
+func NewSecurityGroupsClient(rm RequestMaker) *SecurityGroupsClient {
+	return &SecurityGroupsClient{
+		client:   rm,
+		basePath: &url.URL{Path: "/core/v1/"},
+	}
+}
+
+func (sgc *SecurityGroupsClient) List(
+	ctx context.Context,
+	org OrganizationRef,
+	opts *ListOptions,
+) ([]*SecurityGroup, *katapult.Response, error) {
+	qs := queryValues(org, opts)
+	u := &url.URL{
+		Path:     "organizations/_/security_groups",
+		RawQuery: qs.Encode(),
+	}
+
+	body, resp, err := sgc.doRequest(ctx, "GET", u, nil)
+	resp.Pagination = body.Pagination
+
+	return body.SecurityGroups, resp, err
+}
+
+func (sgc *SecurityGroupsClient) Get(
+	ctx context.Context,
+	ref SecurityGroupRef,
+) (*SecurityGroup, *katapult.Response, error) {
+	return sgc.GetByID(ctx, ref.ID)
+}
+
+func (sgc *SecurityGroupsClient) GetByID(
+	ctx context.Context,
+	id string,
+) (*SecurityGroup, *katapult.Response, error) {
+	u := &url.URL{Path: fmt.Sprintf("security_groups/%s", id)}
+	body, resp, err := sgc.doRequest(ctx, "GET", u, nil)
+
+	return body.SecurityGroup, resp, err
+}
```

Comment:
To keep things more consistent, we're moving towards just always sending lookup refs via query strings. So I'd like to suggest the below changes, which will also require minor updates to the tests.

```suggestion
func (sgc *SecurityGroupsClient) Get(
	ctx context.Context,
	ref SecurityGroupRef,
) (*SecurityGroup, *katapult.Response, error) {
	u := &url.URL{
		Path:     "security_groups/_",
		RawQuery: ref.queryValues().Encode(),
	}
	body, resp, err := sgc.doRequest(ctx, "GET", u, nil)

	return body.SecurityGroup, resp, err
}

func (sgc *SecurityGroupsClient) GetByID(
	ctx context.Context,
	id string,
) (*SecurityGroup, *katapult.Response, error) {
	return sgc.Get(ctx, SecurityGroupRef{ID: id})
}
```